src/main/scala/org/hammerlab/guacamole/commands/StructuralVariantCaller.scala

```diff
@@ -180,6 +182,99 @@ object StructuralVariant {
+
+      val clique = mutable.Set(initPair)
+
+      val graphSize = g.nodes.length
+
+      var cliqueGrow = true
+      while (cliqueGrow && clique.size < graphSize) {
+        cliqueGrow = false
+        // Try to find a node which:
+        // 1. isn't currently in the clique
+        // 2. is connected to every node in the clique
+        // 3. doesn't perturb the SV in a way that makes it incompatible with any reads in the clique
+        mapFirst(edges, (edge: PairedReadGraph#EdgeT) => {
+          val nodes = edge.nodes.toSeq
+          val Seq(node1, node2) = nodes
+          val node1InClique = clique.contains(node1.value)
+          val node2InClique = clique.contains(node2.value)
+
+          // The edge should have one end in the clique and one end out of it.
+          val node = (node1InClique, node2InClique) match {
+            case (false, true) => Some(node1)
+            case (true, false) => Some(node2)
+            case _             => None
+          }
+
+          // TODO: DELLY caches a list of incompatible vertices
+
+          // The new node should be fully connected with the rest of the clique.
+          node filter { n =>
+            {
+              clique.subsetOf(n.neighbors map { _.value })
+            }
+          } filter { n =>
+            // And it should be compatible with the existing SV.
+            {
+              val (_, nodeGapMin, nodeGapMax, _) = n.value.startsAndStops
+              val newSvStart = Math.max(svStart, nodeGapMin)
+              val newSvEnd = Math.min(svEnd, nodeGapMax)
+
+              // The amount of wiggle room could shrink either because of the new read, or because it shrinks the
+              // the induced structural variant and pulls another read closer to being incompatible.
+              val wiggleNewRead = n.value.insertSize - maxNormalInsertSize - (newSvEnd - newSvStart)
+              val wiggleChange = wiggle + (svEnd - svStart) - (newSvEnd - newSvStart)
+              val newWiggle = Math.max(wiggleNewRead, wiggleChange)
+
+              if ((newSvStart < newSvEnd) && (newWiggle <= 0)) {
+                svStart = newSvStart
+                svEnd = newSvEnd
+                wiggle = newWiggle
+                cliqueGrow = true
+                clique += n.value
+                true
+              } else {
+                false
+              }
+            }
+          }
+        })
+      }
+
+      val reads = clique.map(_.value).toSet
+      SVClique(reads=reads,
+               span=GenomeRange(reads.head.read.referenceContig, svStart, svEnd),
+               wiggle=Math.abs(wiggle))
+    }
+
+    def mapFirst[A, B](seq: Seq[A], f: (A) => Option[B]): Option[B] = {
+      seq.view.map(f).find(_.isDefined).flatten
+    }
+
+    def findCliques(g: PairedReadGraph, maxNormalInsertSize: Int): Traversable[SVClique] = {
+      for {
+        c <- g.componentTraverser()
+        if c.nodes.size >= 2
+      } yield findOneClique(c.toGraph, maxNormalInsertSize)
```

Comment:
this feels like it will miss the largest clique in the component a lot of the time. is the heuristic better than I'm giving it credit for?